Frontend/app/class/detail/detail.html

```diff
@@ -19,13 +19,14 @@
           <br/>
         </span>
         <p>{{model.description}}</p>
+        <button class="btn btn-sm btn-primary" (click)="joinClass(classCode)" [ladda]="isJoinClassInprogress" *ngIf="!isStudentInClass">Join Class</button>
```

Comment:
I think you mean `model.code` for `joinClass()` I don't see classCode anywhere in html